### ISSUE TITLE
Support descriptions on seeds, snapshots in .yml files

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#1974]
### Describe the bug
Seeds can be tested in .yml files, but they can not be documented there. While supporting a `description` field for seeds in the manifest might be straightforward, the bigger thing to tackle here is the lack of clarity around the schema.yml spec. Right now, nodes can be defined under one of `sources:` or `models:`.

It's not obvious that seed files (and snapshots) should be defined under `models:`. Should dbt instead support other top-level keys like `seeds:` and `snapshots:`? It's conceivable that each of these resource types would support their own configurations, so separating them out could make parsing the schema.yml file (and surfacing syntax errors to users) simpler. You can also imagine this paradigm extending to document macros.

### Expected behavior
Descriptions on seeds and snapshots should be populated in the rendered documentation site.

**The output of `dbt --version`:**
```
v0.15.0
```

**Comment by clrcrl** [comment]
I like to think of `snapshots`, `seeds` and `models` as `resources`. I think it makes sense to have sources defined differently -- while `sources` are describing a thing that already exists, `resources` is describing a thing that dbt builds. I don't have strong convictions here though!

**Comment by jthandy** [comment]
I just ran into this myself on some work I've been doing. Thanks for documenting this issue.

Like @clrcrl, I also don't have super-strong opinions here, although my initial reaction is to call out the different resource types in the yml tree specifically like you suggested @drewbanin -- they could definitely have different properties in the future. 

The thing that this ties into for me, though, is the `--models` flag. I run a single seed by typing:

```
dbt run --models stg_[seed_name]
```

We've talked for a long time about unifying dbt commands such that `dbt run` actually runs _everything_. I could imagine the selector changing to `nodes` instead of `models`. That's obviously a much bigger topic than this more specific issue, but I do feel like these two topics are related so I wanted to mention it.
